lib/document.js

```diff
@@ -2524,9 +2524,8 @@ Document.prototype.$__validate = function(pathsToValidate, options, callback) {
       // If you `populate()` and get back a null value, required validators
       // shouldn't fail (gh-8018). We should always fall back to the populated
       // value.
-      let pop;
-      if (val == null && (pop = _this.populated(path))) {
-        val = pop;
+      if (val == null && _this.populated(path)) {
+        val = _this.depopulate(path);
```

Comment:
It looks like this would depopulate the path on the document, rather than depopulating just for the validator. We don't want the depopulate to persist.